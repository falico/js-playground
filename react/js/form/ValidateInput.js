import React from 'react';

class ValidateInput extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <label>
        <span>{this.props.label}</span>
        <input
          ref='input'
          type='text'
          value={this.props.value}
          placeholder={this.props.placeHolder}
          onChange={this.props.update} />
      </label>
    )
  }
}

ValidateInput.propTypes = {
  label: React.PropTypes.string,
  value: React.PropTypes.string,
  placeHolder: React.PropTypes.string,
  update: React.PropTypes.func.isRequired,
}

ValidateInput.defaultProps = {
  label: '',
  value: '',
  placeHolder: ''
}

export default ValidateInput
