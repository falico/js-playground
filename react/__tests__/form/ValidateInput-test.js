import React from 'react';
import ValidateInput from '../../js/form/ValidateInput';
import Jest from 'jest';
import TestRenderer from 'react-test-renderer';

test('Validation input renders without error', () => {
  const component = TestRenderer.create(
    <ValidateInput
      label='Input label'
      placeHolder='Input placeholder'
      value="test"
      errorMsg='' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
