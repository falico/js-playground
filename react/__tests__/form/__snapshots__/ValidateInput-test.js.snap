exports[`test Validation input renders without error 1`] = `
<div>
  <label>
    <span>
      Input label
    </span>
    <input
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Input placeholder"
      type="text"
      value="test" />
  </label>
  
</div>
`;
